Work with full HIT parser path

milljm opened the pull request:
Make the full path of the hit parser node tree available to
the tester object. This allows us to understand the path
hierarchy.

Closes #13793

Commit:
Work with full HIT parser path

Make the full path of the hit parser node tree available to
the tester object. This allows us to understand the path
hierarchy.

Closes #13793

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/375044/) on 32bc178 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13829/site)

This comment will be updated on new commits.

Commit:
Fix nested layer tests

Add appropriate prefix to those tests which now require it.
Fix an interesting issue discovered during this process.
- A skipped test, which has failed due to dependencies was not
being detected properly.

Commit:
Updating two test spec files

rebased on next to fix two recently modified test spec files.

milljm commented:
This seems to break Pronghorn and Mammoth. I need to investigate.

milljm commented:
A fix is in place to add the missing Test block to several tests in pronghorn and mammoth.

permcody commented on python/TestHarness/schedulers/Job.py:
```diff
@@ -350,8 +350,8 @@ def getJointStatus(self):
-        # Tester has no status, use a job status instead
-        if self.isTimeout() or self.isNoStatus():
+        # Job has failed, or tester has no status
+        if self.isError() or self.isNoStatus():
```
This change seems unrelated. This is intentional correct?

milljm replied:
That was concerning the comment I made in commit 6bb2201:

Fix an interesting issue discovered during this process.
- A skipped test, which has failed due to dependencies was not
being detected properly.

moosebuild commented:
Job [App tests](https://civet.inl.gov/job/375579/) on e263829 : invalidated by @permcody

Retest Pronghorn and Mammoth

moosebuild commented:
Job [App tests PETSc alt](https://civet.inl.gov/job/375580/) on e263829 : invalidated by @permcody

Retest Pronghorn and Mammoth
